src/main/java/io/cdap/plugin/http/source/common/http/HttpClient.java

```diff
@@ -126,8 +126,16 @@ private CloseableHttpClient createHttpClient() throws IOException {
 
     ArrayList<Header> clientHeaders = new ArrayList<>();
 
-    AuthType authType = config.getAuthType();
+    // oAuth2 - old version
+    if (config.getOauth2Enabled()) {
+      String accessToken = OAuthUtil.getAccessTokenByRefreshToken(HttpClients.createDefault(), config.getTokenUrl(),
+                                                                  config.getClientId(), config.getClientSecret(),
+                                                                  config.getRefreshToken());
+      clientHeaders.add(new BasicHeader("Authorization", "Bearer " + accessToken));
+    }
 
+    // auth check
+    AuthType authType = config.getAuthType();
```

Comment:
Can you just do something like `if (config.getOauth2Enabled()) { authType = OAUTH2; }`? This would reduce code duplication and ensure we don't add the header twice if for whatever reason both configs are set